### ISSUE TITLE
Error : Column 'salt' cannot be null - After updating Symfony

### ISSUE THREAD

**Issue by eved42** [FriendsOfSymfony/FOSUserBundle#2718]
Everything was working well for more than a year and, after updating Symfony, I can't create a new user account.
`SQLSTATE[23000]: Integrity constraint violation: 1048 Column 'salt' cannot be null`

Your bundle works fine with Symfony 3.1 and I use 3.3 now.
I did a composer update.

composer.json 
```
"php": ">=5.5.9",
"doctrine/doctrine-bundle": "^1.6",
"doctrine/orm": "^2.5",
"friendsofsymfony/user-bundle": "~2.0",
"sensio/distribution-bundle": "^5.0.19",
"sensio/framework-extra-bundle": "^3.0.2",
"symfony/assetic-bundle": "^2.8",
"symfony/monolog-bundle": "^3.1.0",
"symfony/polyfill-apcu": "^1.0",
"symfony/swiftmailer-bundle": "^2.3.10",
"symfony/symfony": "3.3.*",
"twig/twig": "^1.0||^2.0"
```

My User entity : 
```
<?php

namespace AppBundle\Entity;

use FOS\UserBundle\Model\User as BaseUser;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity
 * @ORM\Table(name="fos_user")
 */
class User extends BaseUser
{
	/**
	 * @ORM\Id
	 * @ORM\Column(type="integer")
	 * @ORM\GeneratedValue(strategy="AUTO")
	 */
	protected $id;

	/**
	 * @ORM\Column(type="string", length=50)
	 */
	private $lastname;

	/**
	 * @ORM\Column(type="string", length=50)
	 */
	private $firstname;

	public function __construct()
	{
		parent::__construct();
	}

	/**
	 * Get last name
	 *
	 * @return string
	 */
	public function getLastname()
	{
		return $this->lastname;
	}

	/**
	 * Set last name
	 *
	 * @param string $lastname
	 *
	 * @return User
	 */
	public function setLastname($lastname)
	{
		$this->lastname = $lastname;

		return $this;
	}

	/**
	 * Get first name
	 *
	 * @return string
	 */
	public function getFirstname()
	{
		return $this->firstname;
	}

	/**
	 * Set first name
	 *
	 * @param string $firstname
	 *
	 * @return User
	 */
	public function setFirstname($firstname)
	{
		$this->firstname = $firstname;

		return $this;
	}
}
```

I didn't change my config file nor my security file.
I didn't change my User class.
I did the same as in the documentation of Symfony website.

I didn't drop my database schema and recreate it again, should I do it ?

In short, I didn't change anything except update Symfony.

**Comment by stof** [comment]
FOSUserBundle 2 has this column nullable in its schema. So this means you probably updated the bundle without migrating your DB.

Run the `doctrine:schema:validate` command, which will tell you whether your DB is in sync with your ORM metadat.

**Comment by eved42** [comment]
It came from my `fos_user` table which wasn't up to date anymore.
I compared old and new tables with `doctrine:schema:update --force`

It fixed the problem. Thanks.

**Comment by jayesbe** [comment]
@stof is it possible to have a Symfony 2.8 (FOSUserBundle 1.3) app and second Symfony 3.4 (FOSUserBundle 2.0) and access the same DB ?

Can FOSUserBundle 2.0 be made to be compatible for authentication with a 1.3 schema ?

**Comment by feav** [comment]
the salt fiel is use to hash your password. you may not be able to log after set it empty
